app/models/job_execution.rb

```diff
@@ -94,6 +99,14 @@ def error!(exception)
   end
 
   def run!
+    begin
+      @start_callbacks.each(&:call)
+    rescue => e
+      @output.write(e.message)
+      finish
+      return
```

Comment:
We have to return in the rescue, otherwise the body of the job will run, so we can't really re-use that, we could set a flag instead and then skip things if the flag is already set, but that might get a bit hanky...